### DISCUSSION TITLE
Needed Documnet level security restriction based on its attribute value or relative query which returns true, like in supabase

### DISCUSSION THREAD

**Discussion by rahul3v** [appwrite/appwrite#5973] [Ideas]
## High Priority/Important

cloud-beta : I needed  DLS : should be like I can able to restrict each document based on 
current user (can do with roles),

but can' t do based on
its attribute value, based on any relative relationship value/query 
which is not what i geeting 

like in supabase : RLS : you can restrict row access based on "current user value" as well as  column values or any "sql Query" (which returns boolean value, like `exists(Sql query)`, which gives a lot more flexiblity

**Comment by wiledal** [comment]
Related: https://github.com/appwrite/appwrite/discussions/5822

This is not possible.
The current permissions system can cause a heap of issues since they are set client-side, without validation.

The official suggestions for getting around this is:
A) Lock down all permissions, and make your own API endpoints that does the conditional checks on queries, or
B) Use `Functions` to retroactively clean the database after an operation was done by a user, triggered by events

"A" sounds best to me, because working with `Functions` can be a pain, and it would let you have invalid data before the Function overwrites it, and if the Function fails your data is broken.

I'm opting out of the database provided in Appwrite due to these limitations, and the fact that it's a document store that doesn't allow sub-documents, nor join-operations, making it a double clawed hammer.

**Reply by haexmas** [reply]
that's truly a pain in the ass

**Reply by step135** [reply]
Implementation through middleware filters would be so easy to make.
